### PR TITLE
REF: dont pass exception to check_opname

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/extension/base/ops.py [hunk 1]
```diff
@@ -10,10 +10,32 @@
 
 
 class BaseOpsUtil(BaseExtensionTests):
+    series_scalar_exc: type[Exception] | None = TypeError
+    frame_scalar_exc: type[Exception] | None = TypeError
+    series_array_exc: type[Exception] | None = TypeError
+
+    def get_expected_exception(
+        self, op_name: str, obj, other
+    ) -> type[Exception] | None:
```
Comment (jbrockmendel) [inline pandas/tests/extension/base/ops.py:19]: could follow up by allowing authors to specify the exception message here

**Comment by jbrockmendel** [comment]
I have no idea why the scripts test is failing

**Comment by MarcoGorelli** [comment]
sorry about that, will take a look tomorrow - feel free to xfail/remove/ignore for now

**Comment by jbrockmendel** [comment]
troubleshooting this, reverting the edits to tests.extension.base.ops fixes the script failure.  reverting all the edits _other_ than that file does not fix it.  i still dont see any connection between that and the scripts test.

**Comment by mroeschke** [comment]
> troubleshooting this, reverting the edits to tests.extension.base.ops fixes the script failure. reverting all the edits _other_ than that file does not fix it. i still dont see any connection between that and the scripts test.

I don't think the pre-commit check in `check-test-naming` is working correctly.

`pre-commit` isn't catching the violation but `pytest scripts/tests/test_check_test_naming.py` is because it has examples that trigger a `for _file in (Path("pandas") / "tests").rglob("*.py")` check.

**Comment by jbrockmendel** [comment]
so should running the script manually catch a problem in tests.extension.base?

**Comment by jbrockmendel** [comment]
The scripts/tests that are failing are ones that are hard-coded in the parametrization in test_check_test_naming

**Comment by mroeschke** [comment]
This fixes the tests locally for me

```
diff --git a/pandas/tests/extension/base/ops.py b/pandas/tests/extension/base/ops.py
index 86e01dc240..58da0430d5 100644
--- a/pandas/tests/extension/base/ops.py
+++ b/pandas/tests/extension/base/ops.py
@@ -22,7 +22,7 @@ class BaseOpsUtil(BaseExtensionTests):
         # Find the Exception, if any we expect to raise calling
         #  obj.__op_name__(other)
 
-        # The self.foo_bar_exc pattern isn't great in part because it can depend
+        # The self.obj_bar_exc pattern isn't great in part because it can depend
```

I think the test cases `self.foo` matched your comment.

This is besides the fact that `scripts/tests/test_check_test_naming.py` should not be globbing `pandas/tests`

**Comment by jbrockmendel** [comment]
Yep that fixed it.  Thanks.

**Comment by mroeschke** [comment]
```
pylint...................................................................Failed
- hook id: pylint
- duration: 607.66s
- exit code: 8

************* Module pandas.tests.extension.test_arrow
pandas/tests/extension/test_arrow.py:980:11: R1714: Consider merging these comparisons with 'in' by using 'op_name in ('__divmod__', '__rdivmod__')'. Use a set instead if elements are hashable. (consider-using-in)
```

**Comment by jbrockmendel** [comment]
finally green.  thanks for several steps of help

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**